src/Nethermind/Nethermind.Synchronization/FastBlocks/ReceiptsSyncFeed.cs

```diff
@@ -176,15 +176,10 @@ private bool TryPrepareReceipts(BlockInfo blockInfo, TxReceipt[] receipts, out T
                 }
                 else
                 {
-                    Keccak receiptsRoot = new ReceiptTrie(blockInfo.BlockNumber, _specProvider, receipts).RootHash;
-                    if (receiptsRoot != header.ReceiptsRoot)
-                    {
-                        preparedReceipts = null;
-                    }
-                    else
-                    {
-                        preparedReceipts = receipts;
-                    }
+                    IReleaseSpec releaseSpec = _specProvider.GetSpec(blockInfo.BlockNumber);
+                    preparedReceipts = releaseSpec.ValidateReceipts && new ReceiptTrie(releaseSpec, receipts).RootHash != header.ReceiptsRoot 
```

Comment:
we cannot simply not validate receipts - very dangerous - can we discuss more

Reply:
We have to (just for first 1 mln blocks)